File: tests/integration/fixtures/sample.go
package greeter

type Greeter interface {
	Greet(name string) string
}

type Hello struct {
	Prefix string
}

func (h *Hello) Greet(name string) string {
	return h.Prefix + ", " + name + "!"
}

func MakeGreeting(name string) string {
	h := &Hello{Prefix: "Hello"}
	return h.Greet(name)
}

File: tests/integration/fixtures/sample.rs
pub trait Greeter {
    fn greet(&self) -> String;
}

pub struct Hello {
    pub name: String,
}

impl Greeter for Hello {
    fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }
}

pub fn make_greeting(name: &str) -> String {
    let h = Hello { name: name.to_owned() };
    h.greet()
}
